src/server/lib/okta/idx/challenge.ts

```diff
@@ -82,13 +106,31 @@ export const setPasswordAndRedirect = async (
 	expressRes: ResponseWithRequestState,
 	request_id?: string,
 ): Promise<void> => {
-	return await idxFetchCompletion<ChallengeAnswerPasswordBody>({
-		path: 'challenge/answer',
-		body: {
-			stateHandle,
-			credentials: body,
-		},
-		expressRes,
-		request_id,
-	});
+	const [completionResponse, redirectUrl] =
+		await idxFetchCompletion<ChallengeAnswerPasswordBody>({
+			path: 'challenge/answer',
+			body: {
+				stateHandle,
+				credentials: body,
+			},
+			expressRes,
+			request_id,
+		});
+
+	// set the validation flags in Okta
+	const { id } = completionResponse.user.value;
+	if (id) {
+		await validateEmailAndPasswordSetSecurely(id);
+	} else {
+		logger.error(
+			'Failed to set validation flags in Okta as there was no id',
+			undefined,
+			{
+				request_id,
+			},
+		);
+	}
```

Comment:
Does anything rely on this? If the email isn't validated successfully, what happens to the short-lived user session next? What happens to any new user sessions? (Basically, are we happy leaving the error only logged?)

Reply:
one of the cases where it should be theoretically impossible to get to, but helpful if we log an error message

on next sign in they'd be sent the "security email" if they were ever to end up in this scenario